src/software/util/make_enum/make_enum_helpers.hpp

```diff
@@ -0,0 +1,128 @@
+#pragma once
+
+#include <array>
+#include <string_view>
+#include <utility>
+
+/**
+ * Removes all leading and trailing whitespace from a string.
+ *
+ * @param str the string to strip whitespace from
+ *
+ * @return str stripped of all its leading and trailing whitespace
+ */
+constexpr std::string_view stripWhitespace(const std::string_view str)
+{
+    constexpr std::string_view WHITESPACE = " \t\n\f\r\v";
+
+    const size_t str_begin = str.find_first_not_of(WHITESPACE);
+    if (str_begin == std::string::npos)
+    {
+        return "";
+    }
+
+    const size_t str_end   = str.find_last_not_of(WHITESPACE);
+    const size_t str_range = str_end - str_begin + 1;
+
+    return str.substr(str_begin, str_range);
+}
+
+/**
+ * Counts the number of substrings separated by a delimiter.
+ *
+ * @param str a string of substrings separated by the delimiter
+ * @param delimiter the delimiting character between substrings
+ *
+ * @return the number of substrings in the string
+ */
+constexpr size_t countSubstrings(std::string_view str, char delimiter)
+{
+    if (str.empty())
+    {
+        return 0;
+    }
+
+    // There is always at least one substring if the string isn't empty
+    size_t count = 1;
+    for (size_t i = 0; i < str.size(); ++i)
+    {
+        // Trailing delimiters are not counted
+        if (str.at(i) == delimiter && (i != str.size() - 1))
+        {
+            ++count;
+        }
+    }
+    return count;
+}
+
+/**
+ * Splits a string of substrings separated by a delimiter into an array containing
+ * those substrings. All leading and trailing whitespace for each substring will be
+ * removed.
+ *
+ * @param str a string of substrings separated by the delimiter
+ * @param delimiter the delimiting character between substrings
+ * @param index_seq an index sequence 0, 1, 2, ..., (N - 1) where N is the number of
+ * substrings in str (use countSubstrings to count the number of substrings)
+ *
+ * @return an array containing the substrings
+ */
+template <size_t... INDEX_SEQ>
+constexpr auto splitString(std::string_view str, [[maybe_unused]] char delimiter,
```

Comment:
why does delimiter need `[[maybe_unused]`?

Reply:
If INDEX_SEQ is empty then the fold expression is never expanded, so `delimiter` goes unused